frontend/src/Course.js

```diff
@@ -0,0 +1,104 @@
+import Group from './Group.js'
+
+/**
+ * Represents a single course
+ * e.g. Algoritmy a Datové struktury II (přednáška)
+ *
+ * Has many groups. A student can enroll to only a single group.
+ */
+export default class Course {
+  constructor (type, name) {
+    /**
+     * Course id used for list rendering and other id stuff
+     */
+    this.id = type + ';' + name
+
+    /**
+     * Type of the course (lecture/practicals)
+     */
+    this.type = type
+
+    /**
+     * Name of the course (human-readable, in Czech)
+     */
+    this.name = name
+
+    /**
+     * A list of course groups
+     */
+    this.groups = []
+
+    /**
+     * Is this course allowed to be considered during schedule creation?
+     */
+    this.allowed = true
+
+    /**
+     * Course priority with possible values: {1, 2, 3}
+     */
+    this.priority = 2
+  }
+
+  /**
+   * Returns human-readable name of the course type
+   */
+  getTypeName() {
+    switch (this.type) {
+      case 'P': return 'přednáška'
+      case 'X': return 'seminář'
+      default:  return 'jiné'
+    }
+  }
+
+  /**
+   * Creates the server query for creating the schedule
+   */
+  static createScheduleQuery(courses) {
+    const REWARDS = [1, 100, 10000]
+
+    let query = []
+    
+    // maps sent data to application data
+    let queryMap = []
+
+    // courses actually sent to the server
+    let sentCourses = courses.filter(c => c.allowed && c.groups.some(g => g.allowed))
+
+    for (let i = 0; i < sentCourses.length; i++) {
+      let course = sentCourses[i]
+      let sentGroups = course.groups.filter(g => g.allowed)
+
+      query.push({
+        id: course.type + ";" + course.name, // TODO: does the server need this argument?
```

Comment:
Potřebuje server tuhle hodnotu?
Opět tu narážíme na to, jak identifikovat `Course`. Jinak myslím, že ten komentář jsem tam nechal, protože jsem tu hodnotu zkoušel neposílat a stejně mi server bez problémů vyhověl. Ale nepamatuju si to jistě.